src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -513,19 +513,31 @@ private void processCloudWatchAttemptLogInformation(Map<String, Set<String>> com
                                                                 cloudWatchAttemptLogFileInformation) {
         //Todo: eventually the file and fileName should be obtained by the fileHash
         LogFile file = new LogFile(fileName);
-        String componentName = attemptLogInformation.getComponentName();
         boolean isActiveFile = false;
         //TODO: setting this flag is only to develop incrementally without having to changed all tests yet, so that
         // we can avoid a massive PR. This will be removed in the end.
         if (ACTIVE_LOG_FILE_FEATURE_ENABLED_FLAG.get()) {
-            String fileHash = cloudWatchAttemptLogFileInformation.getFileHash();
-            LogFileGroup logFileGroup = attemptLogInformation.getLogFileGroup();
-            // TODO: the following logic is only for passing this small PR while not changing the current context.
-            //  We will grab the fileHash in the future.
-            isActiveFile = logFileGroup.isActiveFile(fileHash);
+            try {
+                String fileHash = cloudWatchAttemptLogFileInformation.getFileHash();
+                LogFileGroup logFileGroup = attemptLogInformation.getLogFileGroup().syncDirectory();
+                Optional<LogFile> file1 = logFileGroup.getFile(fileHash);
+                // TODO: this is temporary for not changing the original logic, will be well handled.
+                if (!file1.isPresent()) {
+                    logger.atDebug().log("The fileHash does not exist in directory");
+                    return;
+                }
+                file = file1.get();
+                fileName = file.getAbsolutePath();
+                // TODO: the following logic is only for passing this small PR while not changing the current context.
+                //  We will grab the fileHash in the future.
+                isActiveFile = logFileGroup.isActiveFile(fileHash);
```

Comment:
Using `isActiveFile` and passing a hash doesn't seem like a good api. Why don't we

```java
LogFile activeFile = logGroup.getActiveFile();
isActiveFile = activeFile.equals(file)
```